openraft/src/replication/mod.rs

```diff
@@ -751,8 +762,9 @@ impl<C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStorage<C>> Replication
             let _en = span.enter();
 
             // Check raft channel to ensure we are staying up-to-date
-            self.try_drain_raft_rx().await?;
-            if self.has_more_log() {
+            let ret = self.try_drain_raft_rx().await;
+            let has_more_logs = self.process_event_result(ret)?;
+            if has_more_logs {
                 // if there is more log, continue to send_append_entries
                 continue;
             }
```

Comment:
There is a chance there is no incoming event but there are logs that need to be sent left in the store.
E.g., when the last `send_append_entries` exceeds config `max_payload_entries`, it should just continue to try sending entries at once.

Reply:
A solution to this problem could be: always read the store to determine the last log id by calling `get_log_state()`.

Reply:
> There is a chance there is no incoming event but there are logs that need to be sent left in the store. E.g., when the last `send_append_entries` exceeds config `max_payload_entries`, it should just continue to try sending entries at once.

I have handled this case in this PR: in the loop if `send_append_entries` return `HasMoreLogs` error, it will call `send_append_entries` directly again.